packages/protocol/contracts/L1/libs/LibProving.sol

```diff
@@ -204,6 +189,34 @@ library LibProving {
         });
     }
 
+    function overwriteForkChoice(
+        TaikoData.State storage state,
+        TaikoData.Config memory config,
+        AddressResolver resolver,
+        uint256 blockId,
+        TaikoData.BlockEvidence memory evidence
+    ) internal {
+        if (msg.sender != resolver.resolve("forkchoice_failsafe", true)) {
+            revert L1_NO_AUTH_TO_OVERWRITE_FK();
+        }
+
+        if (blockId <= state.lastVerifiedBlockId || blockId >= state.numBlocks) {
```

Comment:
Small but not completely trivial logic that is shared with the prover code, maybe a bit better to move this to a util function called isBlockBeingProven or something.

Reply:
Hmm I would expect the function to get inlined. I see the number of runs is also set pretty low: `optimizer_runs = 200` which may impact this. Maybe measure again after increasing that value.

In any case the number of runs used by the optimizer probably should be set much higher in general, it used to make quite a bit of a difference if set to a high value (10,000 or something) for some free gas savings.